sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -185,60 +162,48 @@ private static Object deserializeBody(String value, Type resultType, Type wireTy
     }
 
     /**
-     * Given:
-     * (1). the {@code java.lang.reflect.Type} (resultType) of java proxy method return value
-     * (2). and {@link ReturnValueWireType} annotation value indicating 'entity type' (wireType)
-     * of same REST API's wire response body
+     * Given: (1). the {@code java.lang.reflect.Type} (resultType) of java proxy method return value (2). and {@link
+     * ReturnValueWireType} annotation value indicating 'entity type' (wireType) of same REST API's wire response body
      * this method construct 'response body Type'.
      *
-     * Note: When {@link ReturnValueWireType} annotation is applied to a proxy method, then the raw
-     * HTTP response content will need to parsed using the derived 'response body Type' then converted
-     * to actual {@code returnType}.
+     * Note: When {@link ReturnValueWireType} annotation is applied to a proxy method, then the raw HTTP response
+     * content will need to parsed using the derived 'response body Type' then converted to actual {@code returnType}.
      *
      * @param resultType the {@code java.lang.reflect.Type} of java proxy method return value
      * @param wireType the {@code java.lang.reflect.Type} of entity in REST API response body
      * @return the {@code java.lang.reflect.Type} of REST API response body
      */
     private static Type constructWireResponseType(Type resultType, Type wireType) {
-        Objects.requireNonNull(resultType);
         Objects.requireNonNull(wireType);
-        //
-        Type wireResponseType = resultType;
 
-        if (resultType == byte[].class) {
```

Comment:
Should never get a result type from here, `byte[]` is treated as a non-decodeable type in `isReturnTypeDecodable`.